test/source/mock/google-api-mock.ts

```diff
@@ -120,19 +124,39 @@ export const startGoogleApiMock = async (logger: (line: string) => void) => {
         const id = parseResourceId(req.url!);
         const msgs = new Data(acct).getMessagesByThread(id);
         if (!msgs.length) {
-          throw new HttpClientErr(`MOCK thread not found for ${acct}: ${id}`);
+          throw new HttpClientErr(`MOCK thread not found for ${acct}: ${id}`, 404);
```

Comment:
I was intentionally returning error 400 so that we can notice this. It means we have to either edit the test to not use a resource that's missing in the mock data, or update the mock data. You could try to use another http error if you want that's more obvious. Maybe 501 Not Implemented.

But if you were testing something that was missing intentionally.. that's a different story. Maybe we could have a list of ids where we actually know they are missing (on purpose) and give err 404 for these, but still give other err for the rest.

Reply:
I can't remember exactly why I have changed it to `404`. I think I was testing Gmail API and on that request, it returned 404 for me.

I've checked the tests and seems like they work well with `400` status code.

Reply:
let's see what's CI returns for us.

Reply:
Yes, tests are passed